File: daySeven/menstrualWahala/MenstrualApp.js
const input = require("prompt-sync");
const prompt = input();
const now = new Date();
const year = now.getFullYear();

function endOfCycle(date, daysOfCycleDurationOfCycleDuration){
const result = new Date(date);
result.setDate(result.getDate() + daysOfCycleDuration-1)
return result;
};


function endOfFlowPeriod(date, daysOfFlowDuration){
const result = new Date(date);
result.setDate(result.getDate() + daysOfFlowDuration)
return result;
};



function startOfFertileWindow(date, daysOfCycleDuration){
const result = new Date(date);
result.setDate(result.getDate() + daysOfCycleDuration)
result.setDate(result.getDate() - 16)
return result;
};


function endOfFertileWindow(date, daysOfCycleDuration){
const result = new Date(date);
result.setDate(result.getDate() + daysOfCycleDuration)
result.setDate(result.getDate() - 12)
return result;
};

function safePeriodAfterOvulation(date, daysOfCycleDuration){
const result = new Date(date);
result.setDate(result.getDate() + daysOfCycleDuration)
result.setDate(result.getDate() - 11)
return result;
};

const welcomeMessage = `
WELCOME TO YRSD BOO TING

Kindly enter the following details`;

console.log(welcomeMessage);
let month = Number(prompt("Enter the month of your last cycle(MM): "));
let day = Number(prompt("Enter the day of your last cycle(DD): "));
let daysOfCycleDuration = Number(prompt("How long does your cycle typically last: "));
let daysOfFlowDuration = Number(prompt("How long does your flow typically last: "));
const date = new Date(year, (month-1), (day+1));


let loopCondition = 0
while(loopCondition!=1){
const mainMenuDisplay = `
	What would you like to know?

1) End Date of Cycle
2) End Date of Period Flow
3) Ovulation Window
4) Safe Period
0) Exit`;

console.log(mainMenuDisplay);
let mainMenuNavigator = Number(prompt("Enter your move(1-4):  "));

switch(mainMenuNavigator){
	case 1:
		console.log("The end of your cycle would be ", endOfCycle(date, daysOfCycleDuration));
		break;
	case 2:
		console.log("The predicted end of your flow period would be ", endOfFlowPeriod(date, daysOfFlowDuration));
		break;
	case 3:
		console.log("The start of your Ovulation window would be ", startOfFertileWindow(date, daysOfCycleDuration));
		console.log("The end of your ovulation period would be ",endOfFertileWindow(date, daysOfCycleDuration));
		break;
	case 4:
		console.log("Your safe period after your your Ovulation window would be",safePeriodAfterOvulation(date, daysOfCycleDuration), "to", endOfCycle(date, daysOfCycleDuration));
		break;
	case 0:
		console.log("Good bye, boo.");
		loopCondition = 1;
		break;

	default:
		console.log("INVALID OPTION");
};
};

File: daySeven/menstrualWahala/menstrualapp.py
import datetime
import sys
from menstrualappfunctions import *

now = datetime.datetime.now()

print("WELCOME TO YRSD BOO TING")
print("Kindly enter your details below;\n")
try:
	yyyy = now.year
	mm = int(input("Enter the month your cycle started(Jan=>1): "))
	dd = int(input("Enter the day your cycle started(1-31): "))
	cycle_duration = int(input("\nEnter your average cycle length (21-35 days): "))
	flow_duration = int(input("Enter your average flow length (2-10 days): "))
except ValueError:
	sys.exit("INVALID INPUT")



date_that_cycle_started = datetime.date(yyyy, mm, dd)
try:
	current_menstrual_end_date = get_date_of_current_menstrual_cycle_end_date(date_that_cycle_started, cycle_duration)
	date_of_ovulation_end = get_end_date_of_ovulation(current_menstrual_end_date)
except TypeError:
	print("\nINVALID INPUT")
	sys.exit("Sorry, boo. :(")

loop_condition = 0
while(loop_condition!=1):
	main_menu_display = """\n\t\tWhat would you like to know?\n
1) End date for Current Cycle
2) End date for Period Flow
3) Ovulation Window
4) Safe Days
0) Exit"""

	print(main_menu_display)

	main_menu_navigator = int(input("Enter your move (1-4): "))

	match(main_menu_navigator):
		case 1:
			print("\nYour current cycle will end on the" , current_menstrual_end_date)
		case 2:
			print("\nYour flow day is predicated to end on the" , get_date_of_end_of_flow_day(date_that_cycle_started, flow_duration))
		case 3:
			print("\nYour ovulation start date is predicated to be", get_start_date_of_ovulation(current_menstrual_end_date),"to", get_end_date_of_ovulation(current_menstrual_end_date))
		case 4:
			print("\nSafe Period before Ovulation:", date_that_cycle_started, "to", get_date_of_end_of_flow_day(date_that_cycle_started, flow_duration))
			print("Safe Period after Ovulation:", get_safe_day_start_after_ovulation(date_of_ovulation_end), "to",current_menstrual_end_date)
		case 0:
			print("\nGood bye, boo.")
			loop_condition = 1

		case _:
			print("INVALID OPOTION")